src/lib/replication/log_store/home_raft_log_store.cpp

```diff
@@ -321,20 +321,7 @@ bool HomeRaftLogStore::compact(ulong compact_lsn) {
             append(m_dummy_log_entry);
         }
     }
-
-    m_log_store->flush(to_store_lsn(compact_lsn));
-
-    // this will only truncate in memory, and not on disk;
-    // we rely on resrouce mgr timer to trigger real truncate for all log stores in system;
-    // this will be friendly for multiple logstore on same logdev;
-#ifdef _PRERELEASE
-    if (iomgr_flip::instance()->test_flip("force_home_raft_log_truncate")) {
-        REPL_STORE_LOG(TRACE, "Flip force_home_raft_log_truncate is enabled, force truncation, compact_lsn={}",
-                       compact_lsn);
-        m_log_store->truncate(to_store_lsn(compact_lsn));
-    }
-#endif
-
+    m_log_store->truncate(to_store_lsn(compact_lsn));
```

Comment:
in compact here , we need to truncate , which will update the start_index. flush will not update start_index

Reply:
This is done purposefully. We let resource manager do truncation @yamingk  can confirm this.

Reply:
then , then start_index will not be updated on time, which will probably be wrong to upper layer.
for the view of upper layer , if it compact to lsn n , then the start_index it can see after campact should be lsn + 1.
so , I believe we should do truncate here, and not depend on resource manager

Reply:
This needs to be discussed with @yamingk